### ISSUE TITLE
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

### ISSUE THREAD

**Issue by nailcankara** [shap/shap#2262]
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
Traceback:
  File "home/user/environment/lib/python3.6/site-packages/shap/utils/_general.py", line 87, in potential_interactions
      `if not (i in ignore_inds or np.sum(np.abs(val_other)) < 1e-8):`

ignore_inds is in for example **(array([ 4]),)**  or **(array([ 4, 11, 17]),)** format. 

If ignore_inds has 1 element like **(array([ 4]),)** ,  `i in ignore_inds` works.
If ignore_inds has more than 1 element like **(array([ 4, 11, 17]),)** , `i in ignore_inds` throws **ValueError**. 

Can you update your code please?
https://github.com/slundberg/shap/blob/master/shap/utils/_general.py , **Line 66**
Before :  `ignore_inds = np.where((shap_values_matrix.values.T - shap_values_column.values).T.std(0) < 1e-8)`
Should be :  `ignore_inds = np.where((shap_values_matrix.values.T - shap_values_column.values).T.std(0) < 1e-8)[0]`

**Comment by astrogilda** [comment]
I encounter the same error when trying to use cohorts for plotting the bar plot.

**Comment by suzuki-shm** [comment]
I also encountered this error in the same situation as @astrogilda. 
This error is specific to 0.40.0 and does not occur with 0.39.0.

**Comment by mattphillipsphd** [comment]
Yes I had to do a very similar fix to get `dependence_plot` to work:

In lib/python3.6/site-packages/shap/_explanation.py

Replace `if t == Ellipsis:` with `if np.any(t == Ellipsis):`

Well, then I got to the next bug...

**Comment by nilslacroix** [comment]
Same problem here

**Comment by MingxuanZhangPurdue** [comment]
Same Problem Here
